src/location.rs

```diff
@@ -62,14 +62,11 @@ impl Kml {
         Default::default()
     }
 
-    pub fn parse(context: &Context, content: impl AsRef<str>) -> Result<Self, Error> {
-        ensure!(
-            content.as_ref().len() <= (1024 * 1024),
-            "A kml-files with {} bytes is larger than reasonably expected.",
-            content.as_ref().len()
-        );
+    pub fn parse(context: &Context, content: &[u8]) -> Result<Self, Error> {
+        ensure!(content.len() <= (1024 * 1024), "kml-file too large");
```

Comment:
"*is* too large"

Reply:
The whole expression `(1024 * 1024)` is a result of C macro expansion during Rust conversion. At the very least, parenthesis are unnecessary.